plugins/BEdita/Core/src/Model/Action/ListRelatedObjectsAction.php

```diff
@@ -29,16 +35,86 @@ class ListRelatedObjectsAction extends ListAssociatedAction
     public function execute(array $data = [])
     {
         $data += ['joinData' => true];
-        $result = parent::execute($data);
-        if (!($result instanceof Query)) {
-            return $result;
+
```

Comment:
It is long indeed. :( However, should we approve this kind of approach when listing relationships, we might decide to refactor `ListAssociatedEntitiesAction` in the same fashion, and some logic should become reusable.

I'd plan this other refactor ASAP.